samtranslator/model/api/http_api_generator.py

```diff
@@ -88,6 +102,71 @@ def _construct_http_api(self):
 
         return http_api
 
+    def _add_cors(self):
+        """
+        Add CORS configuration if CORSConfiguration property is set in SAM.
+        Adds CORS configuration only if DefinitionBody is present and
+        APIGW extension for CORS is not present in the DefinitionBody
+        """
+
+        if not self.cors_configuration:
+            return
+
+        if self.cors_configuration and not self.definition_body:
+            raise InvalidResourceException(
+                self.logical_id, "Cors works only with inline OpenApi specified in " "'DefinitionBody' property"
+            )
+
+        # If cors configuration is set to true add * to the allow origins
+        if isinstance(self.cors_configuration, bool):
+            # if cors config is true add Origins as "'*'"
+            properties = CorsProperties()
+
+        elif is_instrinsic(self.cors_configuration):
+            # Just set Origin property. Others will be defaults
+            properties = CorsProperties(AllowOrigins=self.cors_configuration)
```

Comment:
Not sure why we are only setting Origins if its intrinsic.

Reply:
Adding the intrinsic function to property `AllowOrigin` and that is handled in the `OpenApi`. If intrinsic function is present we pass it directly to `x-amazon-apigateway-cors` OpenApi extension. Code: https://github.com/awslabs/serverless-application-model/pull/1381/files#diff-182700364b735df64258a3dcb9c47d77R420-R431